docs/index.rst

```diff
@@ -2,19 +2,47 @@
 ops library API reference
 =========================
 
+The `ops` library is a Python framework for writing and testing Juju charms.
+
+For more about the Charm SDK, see https://juju.is/docs/sdk.
+
+The library provides:
+
+- :ref:`ops_main_entry_point` used to initialise and run your charm.
+- :ref:`ops_module`, the API to respond to Juju events and manage the application.
+- :ref:`ops_pebble_module` low-level API for Pebble in Kubernetes containers.
+- :ref:`ops_testing_module`, the framework for unit testing charms in a simulated environment.
+
+You can write a charm in any way you want, but with the `ops` library you get a
+framework that helps you write consistent readable charms following the latest
+best practices, reuse code via charm libs, and separate typical charm concerns
+--- application state management from integration management or lifecycle
+management from testability.
+
+Whether you are a machine or a Kubernetes charm author, `ops` is the recommended way to go.
+
+
 .. toctree::
    :maxdepth: 2
    :caption: Contents:
 
+
+.. _ops_module:
+
 ops module
 ==========
 
 .. automodule:: ops
    :exclude-members: main
 
 
+.. _ops_main_entry_point:
+
 ops.main entry point
 ====================
```

Comment:
I'd change the title to just `op.main`.

Reply:
That's not part of this PR, although this has been changed recently in #1345

@tonyandrewmeyer wdyt?

Reply:
@tmihoc the challenge here is that there is an `ops.main` that is a module, and there's `ops.main` that's a function (the entry point for the charm, basically). Ideally, people don't know anything about `ops.main` the module, but I think it's worth having the slightly longer name here to avoid confusion with people that do know it exists.

Reply:
OK -- let's keep the entry point bit then, and thanks for clarifying! (I think you might have mentioned that to me in the past but I forgot -- sorry!)